src/mlpack/methods/ann/layer/concat_impl.hpp

```diff
@@ -54,13 +83,58 @@ void Concat<InputDataType, OutputDataType, CustomLayers...>::Forward(
           network[i]);
     }
   }
-  // Vertically concatentate output from each layer.
+
+  // Parameter to store dimensions(rowSize).
+  size_t rowSize, oldColSize, newColSize;
   output = boost::apply_visitor(outputParameterVisitor, network.front());
+
+  newColSize = oldColSize = output.n_cols;
+
+  // Axis is specified and useAxis is true.
+  if (useAxis)
```

Comment:
Looks like we could move this into the constructor, since we only need this once?

Reply:
Yeah sure